### PR TITLE
#39: Update cloudformation templates

### LINKED ISSUE #39: Update CloudFormation template to use NodeJS 20 Lambdas
The CloudFormation template for deploying the Exasol cluster uses four Lambdas using NodeJS 16 to implement custom behavior. AWS supports NodeJS 16 only until July 2024.

So we need to migrate the Lambda source code to NodeJS 20 which requires migration to the new JS module system.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #39

This migrate JS Lambdas to NodeJS 20 and AWS SDK v3 (see [cloudformation_template_diff_1.0.4-2.0.0.patch](https://github.com/user-attachments/files/17161605/cloudformation_template_diff_1.0.4-2.0.0.patch) for the actual changes):
* Fix imports
* Remove unnecessary `.promise()` calls
* Fix base64-encoding of encrypted passwords: KMS function `encrypt()` now returns a `Uint8Array` instead of a `Buffer`

Additionally:
* Cleanup Python script that polls until the cluster is fully started, improve log output
* Remove support for “bring your own license” AMI images which are not supported

### CODE REVIEW

File: scripts/cloudformation_template_v2.0.0.yml
Comment (kaklakariada) [inline scripts/cloudformation_template_v2.0.0.yml:1]: See this patch for comparison between version 1.0.4 & 2.0.0:

[cloudformation_template_diff_1.0.4-2.0.0.patch](https://github.com/user-attachments/files/17161605/cloudformation_template_diff_1.0.4-2.0.0.patch)

File: doc/changes/changes_2.0.0.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# terraform-aws-exasol 2.0.0, released 2024-09-27
+
+Code name: Update CloudFormation template
```
Comment (tomuben) [inline doc/changes/changes_2.0.0.md:3]:
```suggestion
Code name: Update CloudFormation template and new DB version
```

File: scripts/exasol_xmlrpc.py [hunk 2]
```diff
@@ -1,65 +1,94 @@
...
-def create_server(address, username, password):
-    url = CLUSTER_URL.format(username, password, address)
-    server = xmlrpc(url, context=ssl._create_unverified_context())
-    return server
 
-def check_db_started(server):
-    print("Checking if 'exadb' database is running")
-    if not server.db_exadb.runningDatabase():
-        raise Exception("Exasol database is not started!")
+def create_server(address, username, password) -> ServerProxy:
```
Comment (tomuben) [inline scripts/exasol_xmlrpc.py:58]: Shouldn't be the function name rather be `create_client` ?

File: scripts/exasol_xmlrpc.py [hunk 3]
```diff
@@ -1,65 +1,94 @@
...
+    parser.add_argument("--username", type=str, required=True)
+    parser.add_argument("--password", type=str, required=True)
 
     args = parser.parse_args()
-    print("The following arguments are provided: '%s'" % args)
+    logger.info("The following arguments are provided: '%s'" % args)
 
     try:
-        server = create_server(args.license_server_address, args.username, args.password)
+        server = create_server(
```
Comment (tomuben) [inline scripts/exasol_xmlrpc.py:83]: Even though the class name of he returned object  is `ServerProxt`, semantically for me it looks more like a client, so
```suggestion
        client = create_client(
```

**Review by tomuben — APPROVED** [review]

**Review by tomuben — DISMISSED** [review]

**Pull request merged**